homeassistant/components/lock/wink.py

```diff
@@ -20,6 +61,65 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
             add_devices([WinkLockDevice(lock, hass)])
 
 
+    def service_handle(service):
+        """Handler for services."""
+        entity_ids = service.data.get('entity_id')
+        if entity_ids is not None:
+            for lock in hass.data[DOMAIN]['entities']:
+                if lock.entity_id in entity_ids:
+                    if service.service == SERVICE_SET_VACATION_MODE:
+                        lock.set_vacation_mode(service.data.get(ATTR_ENABLED))
```

Comment:
This is a lot of duplication. Split it up: first get the entity ids, then call the service on the found entity ids.